### PR TITLE
Add Skeleton Doots

### PR DESCRIPTION
**Pull request by arkav**
doot doot.

### CODE REVIEW

**Comment by Shrimpstress** [comment]
The rarity of skeletons spawning with trumpets should be increased in my opinion. A 1/4 chance is way too high. That being said, the rarity could possibly be only slightly decreased or the same as now in the spooky dimension. 

This also needs merge conflicts fixed.

File: src/main/java/com/fabriccommunity/spookytime/mixin/MixinAbstractSkeletonEntity.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.Random;
+
+@Mixin(AbstractSkeletonEntity.class)
+public class MixinAbstractSkeletonEntity {
+    @Inject(method = "initEquipment", at = @At(value = "TAIL"))
+    protected void initEquipment(LocalDifficulty ld, CallbackInfo cb) {
+        Random random = new Random();
+        if (random.nextInt(3) == 0) {
```
Comment (Draylar) [inline src/main/java/com/fabriccommunity/spookytime/mixin/MixinAbstractSkeletonEntity.java]: Should probably hike up the rarity to ~1/30 or so. Not sure if there's dimension context in the entity at this stage or not, but if there is, you could have a higher chance only in the spooky dimension (and lower everywhere else).

File: src/main/resources/assets/spookytime/sounds.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "doot": {
+    "category": "players",
+    "sounds": [
+      "spookytime:doot"
+    ],
+    "subtitle": "subtitle.spookytime.doot"
```
Comment (Shrimpstress) [inline src/main/resources/assets/spookytime/sounds.json]: You set the subtitle translation key here, but then never translated it in the `en_us.json`

File: src/main/java/com/fabriccommunity/spookytime/common/SpookySounds.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class SpookySounds {
+    public static final SoundEvent DOOT = register("doot");
+
+    public static void init() {
+        // NO-OP
+    }
+
+    private static SoundEvent register(String name)
+    {
+        return Registry.register(Registry.SOUND_EVENT, new Identifier(SpookyTime.MODID, name), new SoundEvent(new Identifier(SpookyTime.MODID, name)));
```
Comment (Shrimpstress) [inline src/main/java/com/fabriccommunity/spookytime/common/SpookySounds.java]: The Identifiers here should be swapped out to use `SpookyTime.id(name)` instead.

**Review by Shrimpstress — CHANGES_REQUESTED** [review]

**Review by Shnupbups — CHANGES_REQUESTED** [review]
Decrease the spawn rate and this should be merge ready!

File: src/main/java/com/fabriccommunity/spookytime/mixin/AbstractSkeletonEntityMixin.java [hunk 4]
```diff
@@ -18,4 +25,13 @@
     private ItemStack getStack(AbstractSkeletonEntity entity, EquipmentSlot equipmentSlot) {
         return MixinHelpers.getEquippedOrPumpkin(entity, equipmentSlot);
     }
+    
+    @Inject(method = "initEquipment", at = @At(value = "TAIL"))
+    protected void initEquipment(LocalDifficulty ld, CallbackInfo cb) {
+        Random random = new Random();
+        if (random.nextInt(3) == 0) {
```
Comment (liach) [inline src/main/java/com/fabriccommunity/spookytime/mixin/AbstractSkeletonEntityMixin.java]: The random should probably be obtained from the entity itself, and the limit of nextInt should be like 8

**Comment by Shnupbups** [comment]
I've changed the chance to 1 in 50 by default (ignore the name of the commit saying 30) and made it configurable. Should be good to merge now.

**Review by Shnupbups — APPROVED** [review]

File: src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java [hunk 5]
```diff
@@ -38,6 +39,7 @@
 	public static Item SOUL_BOTTLE = register("soul_bottle", new Item(new Item.Settings().group(SpookyTime.GROUP)));
 	public static Item PUMPCOWN_SPAWN_EGG = register("pumpcown_spawn_egg", new SpawnEggItem(SpookyEntities.PUMPCOWN, 8273166, 14912029, new Item.Settings().group(SpookyTime.GROUP)));
 	public static Item PUMPKIN_RING = register("pumpkin_ring", new PumpkinRing(new Item.Settings().group(SpookyTime.GROUP)));
+  public static Item SPOOKY_TRUMPET = register("spooky_trumpet", new SpookyTrumpetItem(new Item.Settings().group(SpookyTime.GROUP)));
```
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java]: Inconsistent indentation here. The other lines use tabs.

**Review by Shrimpstress — APPROVED** [review]

File: src/main/java/com/fabriccommunity/spookytime/registry/SpookySounds.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+public class SpookySounds {
+    public static final SoundEvent DOOT = register("doot");
+
+    public static void init() {
+        // NO-OP
+    }
+
+    private static SoundEvent register(String name)
+    {
```
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/spookytime/registry/SpookySounds.java]: Braces should be on the same line as the method declaration.

File: src/main/resources/assets/spookytime/lang/en_us.json [hunk 7]
```diff
@@ -71,5 +72,6 @@
 	"text.spookytime.loading.costumes": "Creating costumes...",
 	"text.spookytime.loading.witch_water": "Mixing witch water...",
 	"text.spookytime.loading.pumpcowns": "Petting pumpcowns...",
-	"tooltip.spookytime.pumpkin_ring": "That Pumpkin Pie looks awfully delicious..."
+	"tooltip.spookytime.pumpkin_ring": "That Pumpkin Pie looks awfully delicious...",
+  "subtitle.spookytime.doot": "Trumpet Doots Spookily"
```
Comment (Juuxel) [inline src/main/resources/assets/spookytime/lang/en_us.json]: Inconsistent indentation here as well. (Should be a tab)

File: src/main/java/com/fabriccommunity/spookytime/mixin/AbstractSkeletonEntityMixin.java [hunk 8]
```diff
@@ -1,9 +1,20 @@
...
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.Random;
 
 import net.minecraft.entity.EquipmentSlot;
 import net.minecraft.entity.mob.AbstractSkeletonEntity;
```
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/spookytime/mixin/AbstractSkeletonEntityMixin.java]: Not a major issue, but these imports aren't ordered in any way. Consider grouping them like they are in the original code.

**Review by Juuxel — CHANGES_REQUESTED** [review]
Looks good. There are just some minor style issues. Could you reindent the added json files using 4 spaces or tab characters?

**Review by Juuxel — APPROVED** [review]

**Pull request merged**